src/pages/index.js

```diff
@@ -145,54 +136,118 @@ function openModal(modal) {
+    closeModal(evt.target);
+  }
+}
 
-  closeModal(editProfileModal);
+function handleModalEscapeKey(evt) {
+  if (evt.key === "Escape") {
+    closeModal(evt.currentTarget);
+  }
 }
 
-function handleNewPostSubmit(evt) {
-  evt.preventDefault();
+function handleToggleLikeCard(evt, data) {
+  api
+    .changeCardLikeStatus({ id: data._id, isLiked: data.isLiked })
+    .then(() => {
+      evt.target.classList.toggle(cardLikedCSSClass);
+    })
+    .catch(console.error);
+}
 
-  const card = {
-    name: newPostModalCaptionInput.value,
-    link: newPostModalImageLinkInput.value,
-  };
+function handleDeleteCard(cardElement, data) {
+  selectedCard = cardElement;
+  selectedCardId = data._id;
 
-  addCardToCardList(card);
+  openModal(deleteCardModal);
+}
 
-  evt.target.reset();
+//Form submit handlers
+function handleEditProfileSubmit(evt) {
+  function makeRequest() {
+    // `return` lets us use a promise chain `then, catch, finally`
+    return api
+      .editUserInfo({
+        name: editProfileModalNameInput.value,
+        about: editProfileModalDescriptionInput.value,
+      })
+      .then((data) => {
+        profileNameLabel.textContent = data.name;
+        profileDescriptionParagraph.textContent = data.about;
+        closeModal(editProfileModal);
+      });
+  }
 
-  closeModal(newPostModal);
+  handleSubmit(makeRequest, evt);
 }
 
-function addCardToCardList(card, method = "prepend") {
-  const cardElement = getCardElement(card);
-  cardsList[method](cardElement);
+function handleEditProfileAvatarSubmit(evt) {
+  function makeRequest() {
+    // `return` lets us use a promise chain `then, catch, finally`
+    return api
+      .editUserAvatar({ avatar: editProfileAvatarModalImageInput.value })
+      .then((data) => {
+        profileAvatar.src = data.avatar;
+
+        evt.target.reset();
+
+        closeModal(editProfileAvatarModal);
+      });
+  }
+
+  handleSubmit(makeRequest, evt);
 }
 
-function handleModalOutsideClick(evt) {
-  if (evt.target.classList.contains("modal_opened")) {
-    closeModal(evt.target);
+function handleNewPostSubmit(evt) {
+  function makeRequest() {
+    // `return` lets us use a promise chain `then, catch, finally`
+    return api
+      .addCard({
+        name: newPostModalCaptionInput.value,
+        link: newPostModalImageLinkInput.value,
+      })
+      .then((data) => {
+        addCardToCardList(data);
+
+        evt.target.reset();
+
+        closeModal(newPostModal);
+      });
   }
+
+  handleSubmit(makeRequest, evt);
 }
 
-function handleModalEscapeKey(evt) {
-  if (evt.key === "Escape") {
-    closeModal(evt.currentTarget);
+function handleDeleteCardSubmit(evt) {
+  function makeRequest() {
+    // `return` lets us use a promise chain `then, catch, finally`
+    return api.deleteCard({ id: selectedCardId }).then(() => {
+      selectedCard.remove();
+
+      closeModal(deleteCardModal);
+    });
   }
+
+  handleSubmit(makeRequest, evt, deletingDeleteButtonText, true);
```

Comment:
Nice idea for the `deleteCardModal` to enable the button in `finally` using `true` here

Reply:
thanks!